### PR TITLE
Add release process for versioned docs.

### PR DESCRIPTION
**Pull request by bgravenorst**
Add instructions to release the documentation when using versioning.

### CODE REVIEW

File: docs/contribute/doc-release-process.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+---
+description: Release process for documentation that uses a versioning system.
+sidebar_position: 8
+---
+
+# Documentation release process
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:6]:
```suggestion
# Release a new docs version
```

Can we update the title of this page to start with a verb, to match most of the other pages in this section?

File: docs/contribute/doc-release-process.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
+---
+description: Release process for documentation that uses a versioning system.
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:2]:
```suggestion
description: Release a new version of the documentation.
```

Update description to start with verb, to match descriptions for other pages in this section.

File: docs/contribute/doc-release-process.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+sidebar_position: 8
+---
+
+# Documentation release process
+
+The following instructions are for documentation that uses the Docusaurus versioning system.
+
+:::info
+Currently, only the Teku documentation utilizes versioning, supporting just two versions: `stable` and `development`.
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:11]:
```suggestion
Currently, only the Teku documentation uses versioning, supporting just two versions: `stable` and `development`.
```

File: docs/contribute/doc-release-process.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+```bash
+npm run docusaurus docs:version 23.6.3
+```
+
+<!--/tabs-->
+
+This command:
+
+- Copies the full `docs/` directory into a new `version-<VERSION-NUMBER>` directory in the `versioned_docs` directory.
+- Creates a new `versioned_sidebars/version-<VERSION-NUMBER>-sidebars.json` file
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:45]:
```suggestion
- Creates a new `versioned_sidebars/version-<VERSION-NUMBER>-sidebars.json` file.
```

File: docs/contribute/doc-release-process.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+- Appends the new version number to the `versions.json` file.
+
+### 2. Update the `docusaurus.config.js` file
+
+In `docusaurus.config.js`, under `presets` > `classic` > `docs`:
+
+- Update the `lastVersion` to the new version number.
+- Under `versions`, update the current stable version to the new version
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:53]:
```suggestion
- Under `versions`, update the current stable version to the new version.
```

File: docs/contribute/doc-release-process.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+[
+  "23.6.3"
+]
+```
+
+### 4. Delete the previous doc versions
+
+Lastly, we'll cleanup the documentation directory by deleting the previous stable docs. For this example it means:
+
+1. In the `versioned_docs` directory, delete the `version-23.6.2` folder
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:116]:
```suggestion
Lastly, we'll clean up the documentation directory by deleting the previous stable docs. For this example it means:

1. In the `versioned_docs` directory, delete the `version-23.6.2` folder.
```

File: docs/contribute/doc-release-process.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+2. In the `versioned_sidebars` directory, delete the `version-23.6.2-sidebars.json` file.
+
+Create your pull request. You can perform a final check using the Preview link that gets created for your PR.
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:119]:
```suggestion
Create your pull request. You can perform a final check using the preview link generated for your PR.
```

File: docs/contribute/doc-release-process.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
+---
+description: Release process for documentation that uses a versioning system.
+sidebar_position: 8
+---
+
+# Documentation release process
+
+The following instructions are for documentation that uses the Docusaurus versioning system.
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:8]:
```suggestion
The following instructions are for documentation that uses the [Docusaurus versioning system](https://docusaurus.io/docs/next/versioning).
```

File: docs/contribute/doc-release-process.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+:::info
+Currently, only the Teku documentation utilizes versioning, supporting just two versions: `stable` and `development`.
+This means previous releases must be removed as part of the release process.
+:::
+
+A new stable version of the documentation is released when a new version of the software is released.
+In the following example we'll release the `23.6.3` version of the documentation (from `23.6.2`).
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:16]:
```suggestion
In the following steps, we'll release the `23.6.3` version of the documentation (from `23.6.2`) as an example.
```

File: docs/contribute/doc-release-process.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+
+## Steps
+
+You'll need to create a documentation pull request with the following changes:
+
+### 1. Create a new version of the documentation:
+ 
+Create a new documentation version by running the following command:
```
Comment (alexandratran) [inline docs/contribute/doc-release-process.md:24]:
```suggestion
### 1. Create a new version of the documentation
 
Create a new version by running the following command in the documentation repository:
```

**Review by alexandratran — COMMENTED** [review]
Some style feedback (the instructions look good).

File: docs/contribute/doc-release-process.md (hunk 1 above)
Comment (joshuafernandes) [reply docs/contribute/doc-release-process.md:6]: This is a probably a good idea given the versioning audience is a subset of the whole

**Pull request merged**